test/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionControllerSpec.scala

```diff
@@ -1036,6 +1036,17 @@ class PersonalDetailsCollectionControllerSpec extends UnitSpec with MockFactory
       document.select("main a[class='govuk-button']").text() shouldBe messages("multi_option_incorrect_details.button")
     }
 
+    "display the accordion with a link to find your Nino when the findMyNino Feature Flag is set to true" in new Setup(true) {
+
+      val result: Future[Result] = controller.incorrectDetails(completionUrl, 2, failureUrl)(request)
+
+      status(result) shouldBe 200
+      contentType(result) shouldBe Some(HTML)
+      charset(result) shouldBe Some("utf-8")
+
+      val document: Document = Jsoup.parse(contentAsString(result))
+      document.select("#accordion-incorrect-details-content-3 p").get(5).select("a").attr("href") should include("/find-your-national-insurance-number/checkDetails?origin=PDV")
+    }
```

Comment:
I don't think it is a good idea to have both a real AuthConnector instance (line 1098 and a mock AuthConnector instance line 1071) in Setup. I would

1) remove the authConnector at line 1098
2) remove the implicit modifier for messagesApi at line 1096
3) explicitly declare the implicit parameters for the new controller at line 1120 as (mockAuthConnector, dwpMessagesApiProvider, mockViewConfig, ExecutionContext.Implicits.global, messagesApi)

seems to work.

Reply:
Thank you for the instructions above - this has been done